cluster/up.sh

```diff
@@ -25,15 +25,23 @@ if [[ "$KUBEVIRT_PROVIDER" != external ]]; then
     $(kubevirtci::path)/cluster-up/up.sh
 fi
 
-# Deploy CNA
-./cluster/kubectl.sh create -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSIOV}/namespace.yaml
-./cluster/kubectl.sh create -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSIOV}/network-addons-config.crd.yaml
-./cluster/kubectl.sh create -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSIOV}/operator.yaml
-./cluster/kubectl.sh create -f ./hack/cna/cna-cr.yaml
-
-# wait for cluster operator
-./cluster/kubectl.sh wait networkaddonsconfig cluster --for condition=Available --timeout=800s
-
+if [[ "$CNAO_PROVIDER" != external ]]; then
+  # Deploy CNAO
+  ./cluster/kubectl.sh create -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSIOV}/namespace.yaml
+  ./cluster/kubectl.sh create -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSIOV}/network-addons-config.crd.yaml
+  ./cluster/kubectl.sh create -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/${CNAO_VERSIOV}/operator.yaml
+  # Deploy NetworkAddonsConfig
+  ./cluster/kubectl.sh apply -f ./hack/cna/cna-cr.yaml
+  # wait for cluster operator
+  ./cluster/kubectl.sh wait networkaddonsconfig cluster --for condition=Available --timeout=800s
+else # CNAO_PROVIDER == external
+  if [[ -z $(./cluster/kubectl.sh get pod -A -l app=kubemacpool) ]]; then
+    if [[ -z $(./cluster/kubectl.sh get networkaddonsconfig --no-headers -o custom-columns=:.metadata.name) ]]; then
+      ./cluster/kubectl.sh apply -f ./hack/cna/cna-cr.yaml
+    fi
+    ./cluster/kubectl.sh patch networkaddonsconfig cluster --type json -p='[{"op":"replace","path":"/spec/kubeMacPool","value":{}}]'
```

Comment:
It can be that the already deployed networkaddonsconfig has no multus configured (I doubt it though), so we have to patch for multus too just in case.

Reply:
This script is getting bloated and the logic is tailorfit for a single usecase.

This PR adds an option to reconfigure a deployed cluster that happens to have KMP installed. It seems more like a script you want to keep in your home directory for a single setup you happen to have. Do we want to reconfigure external clusters? It won't work if NAC is owned by HCO anyway.

CNAO should not be used in this script to deploy KMP for testing purposes. That completely reverts the dependency... This kind of integration script should be on CNAO "sync with KMP commit and deploy it" and then you can run KMP tests on such a cluster.

We don't want to entangle projects in both directions. And we should be cautious about scripts we include in the repo. People may use them.